### PR TITLE
WEB-6d: Navigation Add/Edit/Workflow

### PR DESCRIPTION
**Pull request by andyleach**

### CODE REVIEW

File: src/NavigationWorkflow.php [hunk 1]
```diff
@@ -22,19 +24,22 @@
  * @see      /src/models/Node.php
  */
 class NavigationWorkflow {
+    public $Tree;
+    public function __construct() {
+        $this->Tree = G::build(ArboristWorkflow::class);
+        $this->Tree->setRoot('/sites/'.$_SERVER['SERVER_NAME']);
```
Comment (tyleruebele) [inline src/NavigationWorkflow.php]: Would it be better to receive an Arborist from the calling Controller?

File: src/controllers/P_NavigationController.php [hunk 2]
```diff
@@ -59,4 +62,122 @@ public function do_list(array $argv = [], array $request = []) {
...
+                ]);
+
+                $Node = $this->Tree->setPath(PencilController::NAVIGATION.'/'.$Node->label)->load()->getFirst();
+                if (is_a($Node, Node::class)) {
+                    G::msg("The navigation has been successfully created", 'success');
+                    $this->_redirect('/P_Navigation/edit/'.$Node->node_id);
+                }
+            }
+        }
+        $this->View->json = $json = json_encode([
```
Comment (tyleruebele) [inline src/controllers/P_NavigationController.php]: This is left over from testing and can be removed.

File: src/controllers/P_TestController.php [hunk 3]
```diff
@@ -206,22 +206,16 @@ public function do_nav() {
...
         ];
         $json = json_encode($json);
-        croak($json);
+        //d($json);
 
         $Nav = new NavigationWorkflow();
 
-        croak($Nav->render(($json)));
-        die($Nav->render(($json)));
+        d($Nav->render(($json)));
```
Comment (tyleruebele) [inline src/controllers/P_TestController.php]: Watch out for un-required functions

File: src/NavigationWorkflow.php (hunk 1 above)
Comment (andyleach) [reply src/NavigationWorkflow.php]: Figured it would be easier to just build it in __construct than to pass it every time.

**Pull request merged**